### ISSUE TITLE
Sequence diagrams

### ISSUE THREAD

**Issue by trosel** [Dletta/grappling-gun#1]
This is to get some initial design thoughts down. I was just picturing the below because of the plug-in-ability of enabling/disabling bots.

![image](https://user-images.githubusercontent.com/1299322/67057591-22b3ce00-f11f-11e9-8880-14477cf9b3fb.png)

```uml
@startuml
title Grappling Gun

actor Bob
control GitterBot
database Gun
control DiscordBot
actor Alice

== Initialization ==

note over GitterBot : Log in
GitterBot -> Gun : Create SEA user
note over DiscordBot : Log in
DiscordBot -> Gun : Create SEA user

GitterBot -> GitterBot : Listen on Gitter channel
DiscordBot -> DiscordBot : Listen on Gitter channel



== Up and Running ==

Bob -> GitterBot : "Hello"
GitterBot -> Gun : Submits message in Gun database
Gun -> DiscordBot : Updates subscribers
DiscordBot -> Alice : Writes message to Discord
Alice -> DiscordBot : "Bye"
DiscordBot -> Gun : Submits message in Gun database
Gun -> GitterBot : Updates subscribers
GitterBot -> Bob : Writes message to Gitter
@enduml
```

http://plantuml.com/starting

@Dletta I know you were thinking of a different and more simple design. Can you explain?

**Comment by Dletta** [comment]
I thought I could handle it in one node process. And at first it worked splendidly from Discord to Gitter. When I got gitter to discord, all of a sudden it wouldn't work anymore

**Comment by trosel** [comment]
@Dletta Yeah, I guess Elixir has ruined me, because I think of everything as actors with their own processes, passing messages haha

Is there anything about the above diagram that you think is mistaken or won't work?

**Comment by Dletta** [comment]
Only the SEA login is not possible in the same process. If you login into one, it logs you out of the other

**Comment by trosel** [comment]
@amark Can you provide your insight on how to handle this?

**Comment by amark** [comment]
@Dletta you could run multiple GUN instances in the same process, where each instance is its own SEA user. Would that help?

**Comment by Dletta** [comment]
Does that work? I thought when I auth in it logs others out?

**Comment by amark** [comment]
only in the same instance!

tho there could be other problems like disk conflicts - but I think I fixed this already.

**Comment by Dletta** [comment]
I’ll check it out

**Comment by trosel** [comment]
@amark 
>you could run multiple GUN instances in the same process, where each instance is its own SEA user

If we want to expose this GUN chat log to users directly (via a GUN chat web app for example), would we create just one extra instance for all other human users then?

**Comment by Dletta** [comment]
@trosel Basically the main instance is the connection to the outside world, the other connections are for the chat

**Comment by trosel** [comment]
@Dletta So something more like this?

![image](https://user-images.githubusercontent.com/1299322/67228060-86334980-f406-11e9-81be-db8c48c1aa6b.png)

**Comment by Dletta** [comment]
Yes, exactly
